solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -61,18 +70,179 @@ def run(site, model, init_time, start, end):
     ...     init_time, start, end)
     """
 
-    *solpos_forecast, resampler, solar_position_calculator = model(
+    *forecast, resampler, solar_position_calculator = model(
         site.latitude, site.longitude, site.elevation,
         init_time, start, end)
 
     if isinstance(site, datamodel.SolarPowerPlant):
         solar_position = solar_position_calculator()
         ac_power = pvmodel.irradiance_to_power(
             site.modeling_parameters, solar_position['apparent_zenith'],
-            solar_position['azimuth'], *solpos_forecast)
+            solar_position['azimuth'], *forecast)
     else:
         ac_power = None
 
     # resample data after power calculation
-    resampled = list(map(resampler, (*solpos_forecast, ac_power)))
+    resampled = list(map(resampler, (*forecast, ac_power)))
     return resampled
+
+
+def run_persistence(observation, forecast, issue_time, index=False):
```

Comment:
For a reference forecast, a primary goal needs to be transparency.  I struggled a bit to understand how this function operates.  Some additional text in the docstring would help, but also the logic for setting the data window seems convoluted.